graph/graph.cpp

```diff
@@ -3,6 +3,11 @@
 /*
  * DAG implementation
  */
+DAG::~DAG() {
+    for(Node* node : nodes)
+        delete node;
+}
```

Comment:
I have already added this in the latest commit to `master` and rebasing this branch locally onto `master` duplicates the code, think it's best to rebase first before merging.

Reply:
Branch rebased onto the newest version of master + unnecessary commit removed